pkg/api/handlers/compat/containers.go

```diff
@@ -169,16 +170,16 @@ func KillContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err = con.Kill(uint(sig))
+	signal := uint(sig)
+
+	err = con.Kill(signal)
 	if err != nil {
 		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrapf(err, "unable to kill Container %s", name))
 	}
 
-	if utils.IsLibpodRequest(r) {
-		// the kill behavior for docker differs from podman in that they appear to wait
-		// for the Container to croak so the exit code is accurate immediately after the
-		// kill is sent.  libpod does not.  but we can add a wait here only for the docker
-		// side of things and mimic that behavior
+	// Docker waits for the container to stop if the signal is 0 or
+	// SIGKILL.
+	if !utils.IsLibpodRequest(r) && (signal == 0 || syscall.Signal(signal) == syscall.SIGKILL) {
```

Comment:
Why would signal=0 stop the container?  I though signal==0 means just check if the PID exists?

Reply:
`signal == 0` in docker terms means default which in turn is SIGKILL

Reply:
That's my understanding too: sig0 is never delivered to the target, it's only used as a check by calling code (for existence, also for EPERM). Good catch.

Reply:
Ok.